docs/execution-providers/EP-Context-Design.md

```diff
@@ -93,4 +93,116 @@ It is hard for Execution Providers to generate the partitioned graph within the
   }
 ```
 
-This API returns the array of pointers for EPContext nodes. Execution Provider needs to implement this interface if it has the requirement to generate the context cache model. Otherwise leave it. It is the Execution Provider's responsibility to create the EPContext nodes with its dependencies (like the context binary file if it's not embed_mode). The OnnxRuntime GraphPartitioner use this interface to get the EPContext nodes and generate the partitioned Onnx model. [code details here](https://github.com/microsoft/onnxruntime/blob/544bdd60730270f49f6a5baafdff54065f626776/onnxruntime/core/framework/graph_partitioner.cc#L646-L750)
\ No newline at end of file
+This API returns the array of pointers for EPContext nodes. Execution Provider needs to implement this interface if it has the requirement to generate the context cache model. Otherwise leave it. It is the Execution Provider's responsibility to create the EPContext nodes with its dependencies (like the context binary file if it's not embed_mode). The OnnxRuntime GraphPartitioner use this interface to get the EPContext nodes and generate the partitioned Onnx model. [code details here](https://github.com/microsoft/onnxruntime/blob/544bdd60730270f49f6a5baafdff54065f626776/onnxruntime/core/framework/graph_partitioner.cc#L646-L750)
+
+
+# EPContext with Weight Sharing
+
+## Weight Sharing in Onnx Domain
+In ONNX, weight sharing refers to multiple ONNX models with external weights pointing to the same external weight file. These models use the same tensor names, allowing them to reference the same tensor data.
+<p align="center"><img width="50%" src="../../images/Onnx_weight_sharing.png" alt="Weight sharing across Onnx models"/></p>
+
+## Weight Sharing in EP Domain with EPContext
+EP weight sharing is enabled using a pre-generated EP context binary/blob. This requires users to generate the context binary offline (AOT) on a Linux x86_64 or Windows x86_64 machine. The EP context binary contains multiple graphs that share the same tensors.
```

Comment:
do we need to call out specific platforms here? ( Linux x86_64/Windows x86_64  ?) 
or that this is simply an offline workflow.

Reply:
Right, it requires offline (AOT), regarding the specific platform, user need to check EP page for more details. Will update that.